File: src/main/java/com/zero_delusions/dev_assignment/core/database/service/InventoryService.java
package com.zero_delusions.dev_assignment.core.database.service;

import com.zero_delusions.dev_assignment.core.database.builder.InventoryQueryBuilder;
import com.zero_delusions.dev_assignment.core.database.table.UserData;
import com.zero_delusions.dev_assignment.core.database.utils.HibernateUtils;
import org.hibernate.Session;

import java.util.List;

public final class InventoryService {
    public static List<UserData> findUsersWithDiamondChestplates(int minCount) {
        return HibernateUtils.executeWithSession((Session session) ->
                InventoryQueryBuilder.query()
                        .withItem("minecraft:diamond_chestplate")
                        .withMinCount(minCount)
                        .inInventory(InventoryQueryBuilder.InventoryType.ARMOUR)
                        .execute(session)
        );
    }
}

File: src/main/kotlin/com/zero_delusions/dev_assignment/core/database/service/InventoryService.kt
package com.zero_delusions.dev_assignment.core.database.service

import com.zero_delusions.dev_assignment.core.database.builder.InventoryQueryBuilder
import com.zero_delusions.dev_assignment.core.database.table.UserData
import com.zero_delusions.dev_assignment.core.database.utils.HibernateUtils
import org.hibernate.Session

object InventoryService {
    fun findUsersWithDiamondChestplates(minCount: Int): List<UserData> {
        return HibernateUtils.executeWithSession { session: Session ->
            InventoryQueryBuilder.query()
                .withItem("minecraft:diamond_chestplate")
                .withMinCount(64)
                .execute(session)
        }
    }
}